maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gittest/src/main/java/org/apache/maven/scm/provider/git/command/checkin/GitCheckInCommandTckTest.java

```diff
@@ -102,6 +105,35 @@ public void testRejectedNonFastForwardPush() throws Exception {
                 "check-in should have been rejected since fast forward was not possible", checkInScmResult.isSuccess());
     }
 
+    @Test
+    public void testCommitNoVerify() throws Exception {
+        GitScmTestUtils.setupRejectAllCommitsPreCommitHook(getWorkingCopy());
+        GitScmTestUtils.setDefaultGitConfig(getWorkingCopy());
+        ScmFileSet addedFile = createWorkspaceChange(getWorkingCopy());
+        CheckInScmResult result;
+        try {
+            result = getScmManager().checkIn(getScmRepository(), addedFile, "Commit with pre-commit hook");
+            assertFalse(
+                    "check-in should have been rejected since pre-push hook rejects all commits", result.isSuccess());
+        } catch (Exception e) {
+            // some providers may use an exception to indicate a failed commit
+
+        }
+        // now try with --no-verify
```

Comment:
Is this two separate tests or does the first somehow depend on the second?

Reply:
for me it makes sense to combine in one test class because if one fails, the other one is probably testing the wrong thing. Therefore both positive and negative testing belong together.

Reply:
Done